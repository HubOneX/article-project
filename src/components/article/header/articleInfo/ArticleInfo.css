.article-info {
  display: flex;
  justify-content: space-between;
  font-size: 16px;
  height: auto;

  & .article-author,
  & .article-date {
    margin: 14px 0 -2px;
    font-size: 15px;

    @media screen and (max-width: 480px) {
      font-size: 12px;
    }
  }
}
