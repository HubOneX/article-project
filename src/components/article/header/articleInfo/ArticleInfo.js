import dateUtils from "../../../../utils/dateUtils";
import "./ArticleInfo.css";

const ArticleInfo = ({ date, author }) => {
  const formattedDate = dateUtils.formatDateForArticle(date);

  return (
    <div className="article-info">
      <p className="article-author">{author ? author : "Anonymous"}</p>
      <p className="article-date">{formattedDate}</p>
    </div>
  );
};

export default ArticleInfo;
