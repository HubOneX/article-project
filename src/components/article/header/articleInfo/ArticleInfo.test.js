import renderer from "react-test-renderer";
import ArticleInfo from "./ArticleInfo";
import { render, screen } from "@testing-library/react";

describe("ArticleInfo", () => {
  it("should match the snapshot", () => {
    const articleInfo = renderer.create(
      <ArticleInfo date="2020-09-03T07:27:59.490" author="Author" />
    );

    expect(articleInfo).toMatchSnapshot();
  });

  it("should render 'Anonymous' if athor is not provided", () => {
    render(<ArticleInfo date="2020-09-03T07:27:59.490" author={""} />);
    const anonymousAuthor = screen.getByText("Anonymous");

    expect(anonymousAuthor).toBeInTheDocument();
  });
});
