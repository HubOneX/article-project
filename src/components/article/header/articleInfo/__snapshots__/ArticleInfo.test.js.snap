// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleInfo should match the snapshot 1`] = `
<div
  className="article-info"
>
  <p
    className="article-author"
  >
    Author
  </p>
  <p
    className="article-date"
  >
    07:27 Sep 3, 2020
  </p>
</div>
`;
